### PR TITLE
Support for cpu_quota and cpu_period

### LINKED ISSUE #766: Add support for --cpu-quota & --cpu-period run flags
Any chance we could add support for the above run flags?
https://docs.docker.com/reference/commandline/run/

### PR DESCRIPTION
**Pull request by shin-**
Fixes #766

### CODE REVIEW

**Comment by shin-** [comment]
#764 needs to be merged for the tests to pass.

File: docker/utils/utils.py [hunk 1]
```diff
@@ -601,6 +602,30 @@ def create_host_config(
...
+        raise TypeError(
+            'Invalid type for cpu_quota param: expected int but'
+            ' found {0}'.format(type(cpu_quota))
+        )
+    elif cpu_quota:
+        if compare_version(version, '1.19') > 0:
+            raise errors.InvalidVersion(
+                'cpu_quota param not supported for API version < 1.19'
+            )
+        host_config['CpuQuota'] = cpu_quota
```
Comment (aanand) [inline docker/utils/utils.py]: Minor thing, but this style of conditional makes me do a double-take - plus we're immediately doing another check in the second branch. I think a single outer check and two inner checks would be more normal.

Also, could we use `version_lt`? I find it much easier to read.

```python
if cpu_quota:
    if not isinstance(cpu_quota, int):
        raise TypeError(...)
    if version_lt(version, '1.19'):
        raise errors.InvalidVersion(...)
    host_config['CpuQuota'] = cpu_quota
```

Same for the `cpu_period` conditional.

**Comment by shin-** [comment]
Should be good now.

**Comment by aanand** [comment]
LGTM

**Pull request merged**